simpleflow/swf/helpers.py

```diff
@@ -97,9 +97,14 @@ def find_activity(history, scheduled_id=None, activity_id=None, input=None):
     """
     Finds an activity in a given workflow execution and returns a callable,
     some args and some kwargs so we can re-execute it.
+
+    :type history: simpleflow.history.History
+    :type scheduled_id: str
+    :type activity_id: str
+    :type input: Optional[dict[str, Any]]
     """
     found_activity = None
-    for _, params in history._activities.items():
+    for _, params in history.activities.iteritems():
```

Comment:
Better stick with `items()` since it's python 3 compatible?

I guess you did that change to avoid building an intermediate list and use a iterator/generator instead, that's it? How does it affect performance/memory in your opinion?